ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/date-time-picker.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  DatePicker,
+  IDateInputProps,
+  IDatePickerProps,
+} from '@blueprintjs/datetime'
+import { DateHelpers } from './date-helpers'
+import { MuiOutlinedInputBorderClasses } from '../theme/theme'
+import TextField, { TextFieldProps } from '@material-ui/core/TextField'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import IconButton from '@material-ui/core/IconButton'
+import CalendarIcon from '@material-ui/icons/Event'
+import Popover from '@material-ui/core/Popover'
+import useTimePrefs from './useTimePrefs'
+
+type DateFieldProps = {
+  value: string
+  onChange: (value: string) => void
+  disabled?: TextFieldProps['disabled']
+  variant?: TextFieldProps['variant']
+  error?: TextFieldProps['error']
+  helperText?: TextFieldProps['helperText']
+  label?: TextFieldProps['label']
+  disableShortcuts?: boolean
+  margin?: TextFieldProps['margin']
+  style?: React.CSSProperties
+  /**
+   * Override if you absolutely must
+   */
+  BPDateProps?: IDatePickerProps
+}
+
+const validateShape = ({ value, onChange }: DateFieldProps) => {
+  if (DateHelpers.Blueprint.commonProps.parseDate(value) === null) {
+    onChange(new Date().toISOString())
+  }
+}
+
+const formatInputValue = (value: string) => {
+  if (value) {
+    const date = DateHelpers.Blueprint.commonProps.parseDate(value)
+    return DateHelpers.Blueprint.commonProps.formatDate(date!)
+  } else {
+    return '' // This allows us to display an empty state without label overlap
+  }
+}
+
+/**
+ * DateTimePicker that combines Mui TextField with BlueprintJs DatePicker
+ * @displayValue - the display text that is shown by the TextField. Anytime a valid date is input into the TextField, the value is updated
+ * @value - the value that holds the date. Often converted into a "TimeShiftedIso" to deal with the fact that Dates don't hold timezone data
+ * @isInternalChange - determines whether the updated value came from inside or outside the component
+ */
+const DateTimePicker = ({
+  value,
+  onChange,
+  disabled,
+  variant,
+  error,
+  helperText,
+  label,
+  disableShortcuts,
+  margin,
+  style,
+  BPDateProps,
+}: DateFieldProps) => {
+  useTimePrefs()
+  const [displayValue, setDisplayValue] = useState(formatInputValue(value))
+  const [isInternalChange, setIsInternalChange] = useState(false)
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
+  const dateIconRef = useRef<HTMLButtonElement | null>(null)
+
+  React.useEffect(() => {
+    validateShape({ onChange, value })
+  }, [])
+
+  React.useEffect(() => {
+    if (isInternalChange) {
+      setIsInternalChange(false)
+    } else {
+      // If the value is updated externally (i.e. a new value is passed into the component),
+      // then we need to update the display value.
+      // This allows us to keep the displayValue and value separate
+      // and type in the TextField without being hijacked
+      setDisplayValue(formatInputValue(value))
+    }
```

Comment:
📓 Before this, I had to deal with either:

1. Updating the calendar view wouldn't update the text field, or
1. The text field value would update as I typed